### ISSUE TITLE
NVDA screen reader announcing unwanted info. after  accepting suggestion.

### ISSUE THREAD

**Issue by balajip881** [microsoft/vscode#194255]
After triggering suggestions, pressing the tab or spacebar causes the screen reader to read out unnecessary information. We accept suggestions by pressing up or down, and only then do we proceed to accept or press space. Therefore, please refrain from announcing anything after the suggestion has been accepted or rejected.
For example, when you open a file in Visual Studio Code, such as "test.js," and type the word "function" followed by pressing the spacebar, listen to the screen reader's voice. It should only announce the word "function" and not provide additional information like the filename, auto-complete, or the entire line.

Currently Screen reader is announcing  file name + auto complete + line in windows 
.
In macbook with  voice over it is announcing  file name + auto complete + entire file name

 solution :
1. Please do not announce anything 
2. or announce  typed word after pressing space 
3.    announce suggestion if it  is accepted. 
4. Please do not announce extra info.

**Comment by balajip881** [comment]
It seems to be related to  https://github.com/microsoft/vscode/issues/140220

**Comment by meganrogge** [comment]
related to #145550, although that issue says it does not reproduce with NVDA, unlike what you have said in this issue. 

We're looking into playing an audio cue when the suggest widget is shown and when it is hidden. @jrieken and I can look into this issue when we do that. #147230

@jooyoungseo and @rperez030 do you work around this in any way ATM?

**Comment by meganrogge** [comment]
Also, would appreciate if @isidorn has any context here.

**Comment by isidorn** [comment]
Thanks for the ping. I do not have additional context. I think we should reach out to the screen reader folks for the fix here. This is not something we are directly controlling.
@leonardder do you also get this behaviour with NVDA and VS Code - that on suggestion accept, too much information is being read? Is this configurable from the NVDA side? Is there something in VS Code that we can do to fix this? Thank you!

**Comment by balajip881** [comment]
@isidorn  I use NVDA screen reader, After every suggestion  rejection or acceptance, NVDA screen reader some how renders and announcing  filename and  auto complete and line content.  There is no setting in NVDA screen reader.

**Comment by leonardder** [comment]
This isn't very easy to fix. It is just the default behavior of every screen reader when the focus changes. When the focus jumps from the suggestions list to the editor, NVDA reads the name, role and states of the editor as the editor gets focus again.

**Comment by balajip881** [comment]
@isidorn  It is very difficult to work because in every line  there will be 4 to 5  suggestions. 
In NVDA, it is  reading out only line content,  i need to press ctrl after every suggestion.
In voice over,  it is reading out entire file, it is becoming impossible to work. It is mandatory for me to work inmacbook.So please try to fix some how.
Please let me know how to stop suggestions  in vscode as a temparory 
solution.thanks.

**Comment by balajip881** [comment]
@isidorn 
Solution : 
1.Do not focus to suggestion widget. Use ARIA live regions for announcing  suggestions so that  that no refocusing happens 
does it solve the issue?
I am not an expert in front end. Please let me know if it works.

**Comment by meganrogge** [comment]
Closing this as it is the screen reader controlling this, not us.

> This isn't very easy to fix. It is just the default behavior of every screen reader when the focus changes. When the focus jumps from the suggestions list to the editor, NVDA reads the name, role and states of the editor as the editor gets focus again.
